sqlglot/dialects/spark2.py

```diff
@@ -101,6 +101,8 @@ def _unqualify_pivot_columns(expression: exp.Expression) -> exp.Expression:
 
 
 class Spark2(Hive):
+    SAFE_DIVISION = False
```

Comment:
Why is this `False`? I think it needs to be `False` only for databricks, see https://github.com/tobymao/sqlglot/pull/2516#issuecomment-1796481170